src/fondant/component/data_io.py

```diff
@@ -135,20 +108,34 @@ def load_dataframe(self) -> dd.DataFrame:
             The Dask dataframe with the field columns in the format (<subset>_<column_name>)
                 as well as the index columns.
         """
-        # load index into dataframe
-        dataframe = self._load_index()
-        for name, subset in self.component_spec.consumes.items():
-            fields = list(subset.fields.keys())
-            subset_df = self._load_subset(name, fields)
-            # left joins -> filter on index
-            dataframe = dd.merge(
-                dataframe,
-                subset_df,
-                left_index=True,
-                right_index=True,
-                how="left",
+        dataframe = None
+        field_mapping = self.manifest.field_mapping
+        for location, fields in field_mapping.items():
+            partial_df = dd.read_parquet(
+                location,
+                columns=fields,
+                index="id",
+                calculate_divisions=True,
             )
 
+            if dataframe is None:
+                # ensure that the index is set correctly and divisions are known.
+                dataframe = partial_df
+            else:
+                dask_divisions = partial_df.set_index("id").divisions
+                unique_divisions = list(dict.fromkeys(list(dask_divisions)))
```

Comment:
Why is this needed? And why are you using `dict.fromkeys`? I would assume we can use `set`?